ddtrace/internal/writer/writer.py

```diff
@@ -378,27 +379,41 @@ def flush_queue(self, raise_exc: bool = False):
     def _flush_queue_with_client(self, client: WriterClientBase, raise_exc: bool = False) -> None:
         n_traces = len(client.encoder)
         try:
-            encoded, n_traces = client.encoder.encode()
-
-            if encoded is None:
-                return
-
-            # Should gzip the payload if intake accepts it
-            if self._intake_accepts_gzip:
-                original_size = len(encoded)
-                # Replace the value to send with the gzipped the value
-                encoded = gzip.compress(encoded, compresslevel=6)
-                log.debug("Original size in bytes: %s, Compressed size: %s", original_size, len(encoded))
-
-                # And add the header
-                self._headers["Content-Encoding"] = "gzip"
-
+            payloads = client.encoder.encode()
+            if not isinstance(payloads, list):
+                n_traces = payloads[1]
+            elif len(payloads) == 1:
+                n_traces = payloads[0][1]
         except Exception:
-            # FIXME(munir): if client.encoder raises an Exception n_traces may not be accurate due to race conditions
+            # FIXME(munir): if client.encoder raises an Exception n_traces may not be accurate
+            # due to race conditions
             log.error("failed to encode trace with encoder %r", client.encoder, exc_info=True)
             self._metrics_dist("encoder.dropped.traces", n_traces)
             return
 
+        if isinstance(payloads, list):
+            for payload in payloads:
+                self._flush_single_payload(payload, client=client, raise_exc=raise_exc)
+        else:
+            self._flush_single_payload(payloads, client=client, raise_exc=raise_exc)
+
+    def _flush_single_payload(
+        self, payload: Tuple[Optional[bytes], int], client: WriterClientBase, raise_exc: bool = False
```

Comment:
```suggestion
        self, encoded: Optional[bytes], n_traces: int, client: WriterClientBase, raise_exc: bool = False
```

think I'd prefer the expansion to happen before calling... maybe just a personal preference 🤷🏻